cookbook/starter-apps/Educhain_pedagogy/Backend/app/api/routes.py

```
@@ -0,0 +1,30 @@
+@router.post("/generate-content", response_model=ContentResponse)
+def generate_content_route(req: ContentRequest):
+    try:
+        result = generate_content(req.topic, req.pedagogy, req.params)
+        return ContentResponse(
+            pedagogy=req.pedagogy,
+            topic=req.topic,
+            content=result
+        )
+    except Exception as e:
+        logger.error(f"Error generating content: {e}")
+        raise HTTPException(status_code=500, detail=str(e))
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)